### ISSUE TITLE
aws_ecs_task_definition cannot access family attribute

### ISSUE THREAD

**Issue by bizmate** [terraform-providers/terraform-provider-aws#2026]
### Terraform Version
```
$ terraform -v
Terraform v0.10.7
```
### Affected Resource(s)
- aws_ecs_task_definition

### Terraform Configuration Files
```hcl
data "template_file" "ecs_task_definition_config" {
  template = "${file("config/ecs-task.json")}"
}

resource "aws_ecs_task_definition" "scenario_planner_app" {
  lifecycle {
    create_before_destroy = true
  }

  family                = "ReportingApp"
  container_definitions = "${data.template_file.ecs_task_definition_config.rendered}"
}

# and in the module i call it to get task definition id
module "app_web_service" {
  source = "github.com/azavea/terraform-aws-ecs-web-service?ref=0.2.0"
  # ....
   task_definition_id             = "${aws_ecs_task_definition.scenario_planner_app.family}:${aws_ecs_task_definition.scenario_planner_app.revision}"

```
### Steps to Reproduce
`TF_LOG=DEBUG && TF_LOG_PATH=~/Documents/ciao.log && terraform plan`

Outputs

```
Error running plan: 1 error(s) occurred:

* module.app_web_service.var.task_definition_id: Resource 'aws_ecs_task_definition.scenario_planner_app' not found for variable 'aws_ecs_task_definition.scenario_planner_app.family'

```

Ideas?

**Comment by apparentlymart** [comment]
Hi @bizmate! Sorry for the strange behavior here.

I think this is the same problem as #1644, where the root cause was the definition JSON being invalid but a bug causing that error to not be displayed prominently.

I'm going to close this one just to consolidate the discussion in the other issue. Thanks for reporting this!

**Comment by bizmate** [comment]
@apparentlymart 
ok so if instead of using a template file, I put the content of the json in 
```
container_definitions = <<DEFINITION
...
DEFINITION
```

I get error

> * aws_ecs_task_definition.scenario_planner_app: ECS Task Definition container_definitions is invalid: Error decoding JSON: json: cannot unmarshal object into Go value of type []*ecs.ContainerDefinition

However, I am taking the task definition directly from the AWS console. Also if I check it at https://jsonformatter.curiousconcept.com/ it tells me it is valid json

**Comment by bizmate** [comment]
@apparentlymart , ok never mind :) it is still a learning process for me. I got now that i only need the content from the container definition from the whole json. Thank you

**Comment by krystan** [comment]
This also helped me as I had a stray comma in my json template, would be good if we could get a json parse error, still happening at v0.11.7

**Comment by rivernews** [comment]
Was getting really confused and wondering why the sudden terraform reports the task definition resource "not found", even if the same terraform code succeed in a previous project.

Then I came across this post, return back and validate my json and boom! Found several trailing commas in my json. Trailing commas is probably the most thing I hate in json.

Anyway just want to say thank you guys for reporting this & this post saved my day! Hope terraform can generate more helpful error message for json parse error.
